bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,424 @@
+    assert_full_setting_registry_output HKEY_CURRENT_USER defaults
+
+    run rdctl create-profile --output reg --hive=hkcu --type=Defaults --from-settings
+    assert_full_setting_registry_output HKEY_CURRENT_USER defaults
+}
+
+@test 'generates registry output for hkcu/locked' {
+    run rdctl create-profile --output reg --hive=HKCU --type=locked --from-settings
+    assert_full_setting_registry_output HKEY_CURRENT_USER locked
+}
+
+assert_check_registry_output() {
+    local mainOutput=$1
+    local bashSideTemp
+    local winSideTemp=
+    local testFile
+    local salt
+    local safePolicyName
+
+    assert_success
+
+    # We need to formulate /tmp as a directory both sides can see.
+    bashSideTemp=$(wslpath -a /tmp)           # e.g.: /mnt/c/tmp
+    winSideTemp=$(wslpath -w "$bashSideTemp") # c:\tmp
+    testFile=test.reg
+    salt=$$
+    # Can't write into ...\Policies\ as non-administrator, so populate a different directory.
+    safePolicyName="fakeProfile${salt}"
+    sed "s/Policies/${safePolicyName}/" <<<"$mainOutput" >"${bashSideTemp}/${testFile}"
+    reg.exe import "${winSideTemp}\\${testFile}"
+    reg.exe delete "HKCU\\Software\\${safePolicyName}\\Rancher Desktop" /f /va
+    rm "${bashSideTemp}/${testFile}"
+}
+
+@test 'validate full-setting registry output on Windows' {
+    if ! is_windows; then
+        skip "Test requires the reg utility and only works on Windows"
+    fi
+    run rdctl create-profile --output reg --hive=HKCU --type=defaults --from-settings
+    assert_check_registry_output "$output"
+}
+
+@test 'validate special-characters' {
+    if ! is_windows; then
+        skip "Test requires the reg utility and only works on Windows"
+    fi
+    run bash -c 'json_with_special_chars | rdctl create-profile --output reg --hive=HKCU --type=defaults --input -'
+    assert_check_registry_output "$output"
+}
+
+@test 'generates registry output from inline json' {
+    run rdctl create-profile --output reg --body '{"application": { "window": { "quitOnClose": true }}}'
+    assert_success
+    assert_output - <<'EOF'
+Windows Registry Editor Version 5.00
+[HKEY_LOCAL_MACHINE\SOFTWARE\Policies]
+[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop]
+[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\defaults]
+[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\defaults\application]
+[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Rancher Desktop\defaults\application\window]
+"quitOnClose"=dword:1
+EOF
+}
+
+@test 'generates plist output from settings' {
+    run rdctl create-profile --output plist --from-settings
+    assert_full_setting_plist_output
+}
+
+@test 'verify plutil is ok with the generated plist output' {
+    if ! is_macos; then
+        skip "Test requires the plist utility and only works on macOS"
+    fi
+    run rdctl create-profile --output plist --from-settings
+    assert_success
+    plutil -s - <<<"$output"
+}
+
+@test "don't need a running app for the rest of this test" {
+    rdctl shutdown
+}
+
```

Comment:
Don't know why we need to do this? We shut down at the end of the test anyways (on the platforms that need it).

There are also several tests earlier in the file that don't need a running app either.

Reply:
This was for dev speedup.... Start the app. Don't wait for it. Do the failure-path tests. Then wait for the app. Then shut it down when we're done with `--full-settings`. I know... too fancy, and I didn't commit it. This was a holdover.